### PR TITLE
PYTHON-4300 Forward comment argument in list_search_indexes

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: test/test_index_management.py [hunk 1]
```diff
@@ -94,6 +99,27 @@ def wait_for_ready(self, coll, name=_NAME, predicate=None):
...
 
+    def test_comment_field(self):
+        # Create a collection with the "create" command using a randomly generated name (referred to as ``coll0``).
+        coll0 = self.db[f"col{uuid.uuid4()}"]
+        coll0.insert_one({})
+
+        # Create a new search index on ``coll0`` that implicitly passes its type.
+        search_definition = {"mappings": {"dynamic": False}}
+        implicit_search_resp = coll0.create_search_index(
+            model={"name": _NAME + "-implicit", "definition": search_definition}
```
Comment (ShaneHarvey) [inline test/test_index_management.py:110]: can you test adding a `comment` to this call too?

File: test/test_index_management.py [hunk 2]
```diff
@@ -94,6 +99,27 @@ def wait_for_ready(self, coll, name=_NAME, predicate=None):
...
+        )
+
+        # Get the index definition.
+        self.listener.reset()
+        coll0.list_search_indexes(name=implicit_search_resp, comment="foo").next()
+        event = self.listener.events[0]
+        self.assertEqual(event.command["comment"], "foo")
+
+
+class TestSearchIndexProse(TestSearchIndexIntegration):
```
Comment (ShaneHarvey) [inline test/test_index_management.py:120]: If this subclasses TestSearchIndexIntegration then won't test_comment_field accidentally run twice?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_index_management.py (hunk 1 above)
Comment (blink1073) [reply test/test_index_management.py:110]: Done

File: test/test_index_management.py (hunk 2 above)
Comment (blink1073) [reply test/test_index_management.py:120]: Created a new base class

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**